app/controllers/questionnaires_controller.rb

```diff
@@ -1,5 +1,44 @@
 class QuestionnairesController < ApplicationController
-  def new
-    render nothing: true
+  def show
+    @firm = current_user.firm
+  end
+
+  def update
+    @firm = current_user.firm
+    @firm.update!(firm_params)
+    redirect_to(new_principal_firm_adviser_path, principal: current_user)
```

Comment:
Can you remove the parens from `redirect_to` and pass the `principal` to the path helper instead.